src/super_gradients/training/models/segmentation_models/unet/unet_decoder.py

```diff
@@ -72,12 +91,21 @@ class UpSumBlock(AbstractUpFuseBlock):
     Fuse features with concatenation and followed Convolutions.
     """
 
-    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, **kwargs):
+    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, fallback_mode: str, **kwargs):
         super().__init__(in_channels=in_channels, skip_channels=skip_channels, out_channels=out_channels)
+        if mode in [UpsampleMode.PIXEL_SHUFFLE.value, UpsampleMode.NN_PIXEL_SHUFFLE.value]:
```

Comment:
We have thee duplicate code blocks 

```
 if mode in [UpsampleMode.PIXEL_SHUFFLE.value, UpsampleMode.NN_PIXEL_SHUFFLE.value]:
            # Check if in_channels is divisible by (up_factor ** 2) for pixel shuffle, else fallback to fallback_mode.
            _in_ch = in_channels / (up_factor**2)
            if _in_ch % 1 == 0:
                in_channels = int(_in_ch)
            else:
                mode = fallback_mode
```

Please consider refactoring these into single function to reduce amount of [unnecessary] duplicated code

Reply:
refactored to `validate_upsample_mode` method [bb14e68](https://github.com/Deci-AI/super-gradients/pull/707/commits/bb14e6877d8e24fb2d7de4d74b91d75b892b496f)